src/util/ChartUtils.ts

```diff
@@ -45,25 +46,31 @@ export function getValueByDataKey<T>(obj: T, dataKey: DataKey<T>, defaultValue?:
   return defaultValue;
 }
 /**
- * Get domain of data by key
+ * Get domain of data by key.
  * @param  {Array}   data      The data displayed in the chart
  * @param  {String}  key       The unique key of a group of data
  * @param  {String}  type      The type of axis
  * @param  {Boolean} filterNil Whether or not filter nil values
  * @return {Array} Domain of data
  */
-export function getDomainOfDataByKey<T>(data: Array<T>, key: string, type: string, filterNil?: boolean) {
-  const flattenData = _.flatMap(data, entry => getValueByDataKey(entry, key));
+export function getDomainOfDataByKey<T>(
+  data: Array<T>,
+  key: DataKey<T>,
+  type: BaseAxisProps['type'],
+  filterNil?: boolean,
+): NumberDomain | CategoricalDomain {
+  const flattenData: unknown[] = _.flatMap(data, (entry: T): unknown => getValueByDataKey(entry, key));
 
   if (type === 'number') {
-    const domain = flattenData.filter(entry => isNumber(entry) || parseFloat(entry));
+    // @ts-expect-error parseFloat type only accepts strings
+    const domain: number[] = flattenData.filter(entry => isNumber(entry) || parseFloat(entry));
 
     return domain.length ? [_.min(domain), _.max(domain)] : [Infinity, -Infinity];
```

Comment:
Nit: We could extract [Infinity, -Infinity] as a constant, DEFAULT_DOMAIN

Random side note: 
This line has an interesting edge case, i.e. when the data contains only a single data point. Then the resulting domain is [x, x] which results in a very weird axis and chart.